### PR TITLE
Create entry point

### PR DESCRIPTION
**Pull request by tarikbouari**
In this milestone I implement the following action:

- Create class app
- List all books.
- List all people.
- Create a person (teacher or student, not a plain Person).
- Create a book.
- Create a rental.
- List all rentals for a given person id.
- Create the app entry point in the main.rb file 
- Refactor code and fix linters errors

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+  case option
+  when 1
+    create_student
+  when 2
+    create_teacher
+  else
+    puts 'Invalid choice'
+  end
+end
```
Comment (alexander16108) [inline app.rb]: you have an issue when creating a person `teacher`  currently it shows this error

image

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  @p = Object.new
+  @people.each do |person|
+    @p = person if person.id == person_id
+  end
+
+  rental = Rental.new(date, @b, @p)
+  @rentals << rental
+
+  puts "Rental created successfully\n-------------------------------------"
+end
```
Comment (alexander16108) [inline app.rb]: you have an issue when creating a rental currently your app shows these errors

image

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @tarikbouari , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+  puts 'Enter id of a person'
+  person_id = gets.chomp.to_i
+
+  @rentals.each do |rental|
+    if rental.person.id == person_id
+      puts "On #{rental.date} the book:  #{rental.book.title} is rented by #{rental.person.name}"
+      puts '****************'
+    end
+  end
+end
```
Comment (thecodechaser) [inline app.rb]: - [ ] When I'm trying to list rentals for a person, I get nothing on the screen. Please fix this issue.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @tarikbouari👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @tarikbouari, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

✔️ No linter error.
✔️ Proper Gitflow used.
✔️ Descriptive PR title & description.
✔️ Descriptive readme file.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**